src/main/java/uk/gov/companieshouse/charges/data/config/ChargesApplicationConfig.java

```diff
@@ -51,6 +55,22 @@ public InternalApiClient internalApiClient() {
         return ApiSdkManager.getPrivateSDK();
     }
 
+    @Bean
+    public Supplier<InternalApiClient> internalApiClientSupplier(
+            @Value("${chs.kafka.api.key}") String apiKey,
+            @Value("${chs.kafka.api.endpoint}") String apiUrl) {
+        return () -> {
+            InternalApiClient internalApiClient = new InternalApiClient(new ApiKeyHttpClient(apiKey));
```

Comment:
Can you add the request id to the ApiKeyHttpClient object here?

Reply:
Resolved in latest commit